### ISSUE TITLE
Unable to run 

### ISSUE THREAD

**Issue by mognc** [explainingai-code/StableDiffusion-PyTorch#12]
Hello there, I am trying to run conditional text part and have followed all the instructions but at the end I am facing follow error. Screenshot attached below. It says "Model checkpoint celebhq/ddpm_ckpt_text_cond_clip.pth not found"
<img width="700" alt="Screenshot 2024-03-28 144732" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/0e2ad3f3-5911-4218-a2e3-9e0040f1230a">

**Comment by explainingai-code** [comment]
Hello @mognc , 
When you ran `train_ddpm_cond.py` , what was the configuration , you used. If it was `config/celebhq_text_cond.yaml`. then this would have created a checkpoint in the `celebhq/ddpm_ckpt_text_cond_clip.pth` location.
Can you let me know which config you ran the training with, and if you have changed some parameters, then could you attach that config as well?

**Comment by mognc** [comment]
I didn't changed any parameter and yes I used config/celebhq_text_cond.yaml

**Comment by explainingai-code** [comment]
Okay, then you can you check what is the name of checkpoint file that was created in the `celebhq` folder

**Comment by mognc** [comment]
Sorry, but there is no celebhq folder here. 
<img width="230" alt="Screenshot 2024-03-28 161325" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/684d3ec2-a625-4a7b-8c99-9824c0df3a41">

**Comment by explainingai-code** [comment]
But there should be a celebhq folder after you run the autoencoder. I am assuming you ran `train_vqvae` with the same config file right ?

**Comment by mognc** [comment]
Sorry for troubling you, I missed that block. Its downloading different models atm and I hope my error will be resolved now. Thanks for helping. I am new to this image generation thing so I make silly mistakes.

**Comment by explainingai-code** [comment]
No problem at all @mognc :)
So basically you will first train autoencoder (`train_vqvae.py`) and then you can choose to train unconditional or conditional .
And just make sure to you use the same config file for both the stages(autoencoder and ldm).

Will keep the issue open for now, and you can close it once you have successfully ran the text conditional training.
Feel free to comment here when you run into any further problems during that.

**Comment by mognc** [comment]
Unfortunately the error did not get resolved, I am sharing attachments showing the commands I used also showing my celebhq folder contents.
<img width="557" alt="Screenshot 2024-03-28 180307" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/9826d1d3-f1ac-48d9-8db1-44debe7e1322">
<img width="529" alt="Screenshot 2024-03-28 180248" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/3f56120f-910d-477c-93e1-190b2772ce4a">
<img width="470" alt="Screenshot 2024-03-28 180203" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/3879b97f-dfac-4bed-94f6-e22b625ed79a">

**Comment by explainingai-code** [comment]
After running train_vqvae , did you run the train_ddpm_cond ?  Did that fail ?

**Comment by mognc** [comment]
After running this command "!python -m tools.train_vqvae --config config/celebhq_text_cond.yaml" it displayed the training is completed. Then i ran "!python -m tools.sample_ddpm_text_cond --config config/celebhq_text_cond.yaml" which failed. I also have attached pics above for your reference.

**Comment by explainingai-code** [comment]
Yes but the train vqvae is only for Stage I. This will only train the autoencoder and not the diffusion model.
Once auto encoder is trained, we need to run train_ddpm_cond for training a conditional latent diffusion model.
And after that is trained, you would be able to generate data using sample_ddpm_text_cond command

**Comment by mognc** [comment]
Hey there, my model got in running state yesterday but the output was not that I desired. And I am bit confused what changes I have to make to improve them. I have 100 images dataset with 100 captions file. The dimensions of pics are 600*800. I have made dataset and yaml file similar to "celebhq.yaml" and celeb_dataset.py file. I also have modified the targets to my own files. I want to generate similar pics as my dataset but with variations in them. I figured out to achieve this task I will be using Training text conditional LDM method and will be creating similar file as "celebhq_text_cond.yaml" file right?. But now I am unsure what additional changes I have to make in my new three files I have created that will help me achieve my goal like can you pin out parameters I have to change like no of epochs I have to train and etc. Also I have enables save latents in those file as you mentioned in readme file to speed up the training process.

**Comment by explainingai-code** [comment]
How many epochs/steps did you train the autoencoder for ? and could you add some output examples of autoencoder.
Same for LDM. Because that will help me understand which stage is not generating output as desired.

The first thing I would suggest is have more images,  maybe 2K images to start with.
Second is there a reason you want to formulate this as a text conditional problem and not class conditional problem ? This is because for text conditional problems you would be training additional cross attention layers and the training will also be slower. So if you can achieve your goal by just formulating this as a class conditional problem , I would suggest to first try that.

**Comment by mognc** [comment]
Well I didn't change epochs or samples. At the moment I don't have access to those outputs as Collab erase all the data after session terminates. But the ldm stage was not producing correct output it was just blur pic.
Well I assumed to add variations to a dataset through prompt I will need to use text condition and not class condition. I might be wrong as I am no expert. But the final goal is to add variations according to the user prompt. I have dataset of walls and the user will enter prompt like snow on walls or shadow on the walls and the relevent pic will be generated.

**Comment by explainingai-code** [comment]
If you didnt change any parameters then that means the autoencoder ran for only 20 epochs and the discriminator didnt even start because the config has the start of discriminator at 15000 steps.
So you should anyways train autoencoder again and change the discriminator_start parameter to the number of steps after which you start seeing decent but blurry outputs from autoencoder.

For the conditioning, if all you have are texts of type '<obj> on walls' where obj can be one of K things, then you can use class conditioning with K classes rather than text conditioning.

**Comment by mognc** [comment]
Ok I will change that parameter. And I will try conditioning too, but I just want to make sure like my dataset is simple it don't include type of variations I want like snow or dust. This will not be a problem right ?

**Comment by explainingai-code** [comment]
Also the ldm epochs are set at 100 epochs but this was for celebhq dataset with 30000 images.
I would suggest in the current setting with 100 images, you should train till 1000 epochs to validate quality of ldm outputs(more if you see that the quality is still continuing to improve )

**Comment by explainingai-code** [comment]
"but I just want to make sure like my dataset is simple it don't include type of variations I want like snow or dust" 
I didnt get this part. Could you clarify a bit ? Do you mean that you want the model to generate variations for which you don't have images ?

**Comment by mognc** [comment]
Yes, I don't have pics of variations I want.

**Comment by explainingai-code** [comment]
But if the model has never seen what 'snow' looks like anytime during the training, it will not be able to generate 'snow on walls' right ?

**Comment by mognc** [comment]
Well my friend used some pre trained models and those were producing results, so I am not sure of this model how it works. So should I add simple pics of snow, dust and other variations and merge them with walls dataset?

**Comment by explainingai-code** [comment]
Yes pre-trained model would work because that has seen what 'snow' looks like. But this model will be trained from scratch.
So I would suggest to either use the pre-trained model and then fine tune it using libraries like diffusers. 
OR if you want to train using this repo from scratch then add in those images for training.

**Comment by mognc** [comment]
Well I will stick with this repo and will add variations pic along with captions and will merge all dataset together. Thanks for removing all confusions really appreciate your time.

**Comment by mognc** [comment]
I cant figure out this part of  guide like what to change and where 
<img width="607" alt="image" src="https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/ae923e43-4ee0-487c-be1a-d63ca00c23dc">

**Comment by mognc** [comment]
Hello again, I was unable to train my model class conditionally was unable to solve all errors after changing some files. So i tried training it text conditionally and here are some outputs. 
![current_autoencoder_sample_193](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/e6c6add1-1d20-45ae-a15a-d2e555e78c19)
This is final picture that was generated while training autoencoder at 500 epochs and disc_start at 100.
![x0_996](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/9cae81b4-f04f-4100-b4bd-390f1116bb46)
This is sample generated at x0_996.
![x0_0](https://github.com/explainingai-code/StableDiffusion-PyTorch/assets/76902280/832e1067-5f14-4e21-b238-776c2cc555b7)
And this is sample generated at x0_0. I trained model for 1000 epochs can this result be improved if I train for more epochs and final question does my training file which is ddpm_ckpt_text_cond_clip.pth overwrite every time I run "!python -m tools.train_ddpm_cond --config config/celebhq_text_cond.yaml" cell and new file is saved right ?

**Comment by mognc** [comment]
This is my config file which I edited:

**Comment by explainingai-code** [comment]
I think it would benefit by training the autoencoder more. Specifically two changes:
1. autoencoder_epochs:1000
2. disc_start : 200 x (Number of steps in one epoch)


Basically train for longer and start discriminator after your autoencoder generates the best reconstructions it can. The disc_start is the number of steps after which discriminator should start.

**Comment by explainingai-code** [comment]
Yes the ddpm_ckpt_text_cond_clip.pth is overwritten every time you run the training.
